### PR TITLE
Castle screen, Creature separation window, the sentence "Fast separation into slots:" and the graphic "2" are not displayed when there are only 2 creatures in the starting slot during the distribution between the hero's army and the garrison

### LINKED ISSUE #3621: Castle screen, Creature separation window, the sentence "Fast separation into slots:" and the graphic "2" are not displayed when there are only 2 creatures in the starting slot during the distribution between the hero's army and the garrison
In fheroes2, the sentence "Fast separation into slots:" and the graphic "2" are not displayed when there are only 2 creatures in the starting slot during the distribution between the hero's army and the garrison.

In fheroes2:

Creature separation window 001

Creature separation window 002

https://user-images.githubusercontent.com/43583994/120869999-13a1be00-c598-11eb-9aaf-814309d2a729.mp4

### PR DESCRIPTION
**Pull request by felix642**
fix #3621

### CODE REVIEW

File: src/fheroes2/army/army_bar.cpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ void RedistributeArmy( ArmyTroop & troopFrom, ArmyTroop & troopTarget, Army * ar
             ++freeSlots;
 
         const uint32_t maxCount = saveLastTroop ? troopFrom.GetCount() - 1 : troopFrom.GetCount();
-        uint32_t redistributeCount = isSameTroopType ? 1 : troopFrom.GetCount() / 2;
+        uint32_t redistributeCount = isSameTroopType - 1 ? 1 : troopFrom.GetCount() / 2;
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.cpp:63]: The `isSameTroopType - 1` operation is done on boolean type which we should not allow. Could you please explain what was the original intention?

**Review by ihhub — COMMENTED** [review]
Hi @felix642 , I left one question. Could you please answer it?

File: src/fheroes2/army/army_bar.cpp (hunk 1 above)
Comment (felix642) [reply src/fheroes2/army/army_bar.cpp:63]: That was a mistake. I was suppose to add the -1 to line 62. I'll revert this change right now!

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
Hi @Branikolog , could you please test this chnage too?

**Comment by Branikolog** [comment]
Hi, @ihhub.
Hm... I was quite busy last days and I missed this issue....
Why do you want to remove [2] button from this dialog? It could be really useful, if player changed creature quantity in a box, but later changed his mind and decided to split stack into two same size. By pressing [2] player can easily achieve this, rather than scrolling or entering the exact number again...

**Comment by ihhub** [comment]
> Hi, @ihhub.
> Hm... I was quite busy last days and I missed this issue....
> Why do you want to remove [2] button from this dialog? It could be really useful, if player changed creature quantity in a box, but later changed his mind and decided to split stack into two same size. By pressing [2] player can easily achieve this, rather than scrolling or entering the exact number again...

Hi @Branikolog , I thought the issue stated that we should add 2, not to remove. Correct me please if I'm wrong.

**Comment by Branikolog** [comment]
@ihhub 
Sorry for my misunderstanding! 
I need a vacation, probably. :)
I'll retest the changes!

**Comment by ihhub** [comment]
@felix642 , thank you so much for fixing this issue!

**Pull request merged**